ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/util/BeaconStateUtil.java

```diff
@@ -172,32 +168,26 @@ public static BeaconStateWithCache get_initial_beacon_state(
     }
     List<List<Integer>> shuffling =
         get_shuffling(seed, state.getValidator_registry(), shuffling_epoch);
-    UnsignedLong offset = slot.mod(UnsignedLong.valueOf(SLOTS_PER_EPOCH));
-    UnsignedLong committees_per_slot =
-        committees_per_epoch.dividedBy(UnsignedLong.valueOf(SLOTS_PER_EPOCH));
+    long offset = slot % SLOTS_PER_EPOCH;
+    long committees_per_slot = committees_per_epoch / SLOTS_PER_EPOCH;
     // TODO: Double check the order of operations here. The spec may be ambiguous.
-    UnsignedLong slot_start_shard =
-        shuffling_start_shard
-            .plus(committees_per_slot)
-            .times(offset)
-            .mod(UnsignedLong.valueOf(Constants.SHARD_COUNT));
+    long slot_start_shard =
+        shuffling_start_shard + committees_per_slot * offset % Constants.SHARD_COUNT;
```

Comment:
There is still an OOO glitch here as well. This should be `long slot_start_shard = (shuffling_start_shard + committees_per_slot * offset) % Constants.SHARD_COUNT;` according to the spec. The spec is even a little ambiguous as well (committees_per_slot * offset is calculated before adding shuffling_start_shard, even though the addition term is written first), so this might be something we want to clarify with DR. @cemozerr. can you look at this? Last line above the return here: https://github.com/ethereum/eth2.0-specs/blob/v0.5.1/specs/core/0_beacon-chain.md#get_crosslink_committees_at_slot